internal/plugin/server.go

```diff
@@ -375,20 +382,14 @@ func (plugin *NvidiaDevicePlugin) getAllocateResponse(requestIds []string) (*plu
 // and assumes that an MPS control daemon has already been started.
 func (plugin NvidiaDevicePlugin) updateResponseForMPS(response *pluginapi.ContainerAllocateResponse) {
 	// TODO: We should check that the deviceIDs are shared using MPS.
-	for k, v := range plugin.mpsDaemon.Envvars() {
-		response.Envs[k] = v
-	}
+	response.Envs["CUDA_MPS_PIPE_DIRECTORY"] = plugin.mpsDaemon.PipeDir()
 
 	resourceName := plugin.rm.Resource()
 	response.Mounts = append(response.Mounts,
 		&pluginapi.Mount{
 			ContainerPath: plugin.mpsDaemon.PipeDir(),
 			HostPath:      plugin.mpsHostRoot.PipeDir(resourceName),
 		},
-		&pluginapi.Mount{
-			ContainerPath: plugin.mpsDaemon.PipeDir(),
-			HostPath:      plugin.mpsHostRoot.LogDir(resourceName),
-		},
```

Comment:
Looks like this was actually buggy, which is why we probably saw that `log` file I questioned when debugging today.

Reply:
Could be.

Reply:
Were we not looking in the pipe directory though?

Reply:
I don't remember anymore. In any case, the code looks correct now.